exercises/sum-of-multiples/example.rs

```diff
@@ -0,0 +1,17 @@
+use std::collections::BTreeSet;
+
+pub fn sum_of_multiples(limit: u32, factors: &Vec<u32>) -> u32 {
+    let mut multiples: BTreeSet<u32> = BTreeSet::new();
+
+    for f in factors {
+        let mut multiplier = 2;
+        let mut x = f.clone();
```

Comment:
Trying to think about what looks best here, given https://github.com/exercism/xrust/pull/231#discussion_r91913638 . I was thinking about `*f` instead, or a `for &f in factors` followed by a plain `let mut x = f` here. In the end, I don't really care, just pick.